submissions/Levi123/arcade-game/app.js

```diff
@@ -0,0 +1,154 @@
+    positionX: -100,
+    positionY: {
+        firstRowEnemy: DEFAULT_MAP.firstRow + 60,
+        secondRowEnemy: DEFAULT_MAP.secondRow + 60,
+        thirdRowEnemy: DEFAULT_MAP.thirdRow + 60,
+    },
+}
+
+let {firstRowEnemy, secondRowEnemy, thirdRowEnemy} = DEFAULT_ENEMY.positionY;
+
+class Player {
+    constructor(x = DEFAULT_PLAYER.positionX, y = DEFAULT_PLAYER.positionY){
+        this.sprite = DEFAULT_PLAYER.sprite;
+        this.x = x;
+        this.y = y;
+    }
+
+    update(){
+        if (this.y < DEFAULT_MAP.waterLine){
+            setTimeout(() => {
+                this.x = DEFAULT_PLAYER.positionX;
+                this.y = DEFAULT_PLAYER.positionY;
+            }, 300)
+        }
+    }
+
+    render(){
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+
+    handleInput(key){
+       if (key == 'left' && this.x > DEFAULT_MAP.waterLine){
+        this.x -= DEFAULT_MAP.widthBlock;
+       }
+       if (key == 'right' && this.x < DEFAULT_PLAYER.positionY){
+        this.x += DEFAULT_MAP.widthBlock;
+       }
+       if (key == 'up' && this.y > DEFAULT_MAP.waterLine){
+        this.y -= DEFAULT_MAP.heightBlock;
+       }
+       if (key == 'down' && this.y < DEFAULT_PLAYER.positionY){
+        this.y += DEFAULT_MAP.heightBlock;
+       }
+    }
+}
+
+let player = new Player();
+
+class Enemy {
+    constructor(x, y){
+        this.sprite = DEFAULT_ENEMY.sprite;
+        this.x = x;
+        this.y = y;
+        this.speed = this.changeSpeed();
+    }
+
+    changeSpeed(min = DEFAULT_ENEMY.minSpeed, max = DEFAULT_ENEMY.maxSpeed){
+        return Math.floor(Math.random() * (max - min)) + min;
+    }
+
+    enemyKillPlayer(){
+        if(player.x < this.x + DEFAULT_ENEMY.widthEnemy &&
+            player.x + DEFAULT_ENEMY.widthEnemy > this.x &&
+            player.y < this.y + DEFAULT_ENEMY.heightEnemy &&
+            DEFAULT_ENEMY.heightEnemy + player.y > this.y){
+                player.x = DEFAULT_PLAYER.positionX;
+                player.y = DEFAULT_PLAYER.positionY;
+        }
+    }
+
+    update(delta){
+        this.x += this.speed * delta; 
+
+        if(this.x > DEFAULT_MAP.widthMap){
+            this.speed = this.changeSpeed();
+            this.x = DEFAULT_ENEMY.positionX;
+        }
+        this.enemyKillPlayer();
+    }
+
+    render(){
+        ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+    }
+};
+
+let allEnemies = [];
+//If you want create Enemy, add row number (along which a new enemy will run) to array.
+let createEnemies = [1,2,3]
+
+createEnemies.forEach((valueY) => {
+    if (valueY === 1){
+        enemy = new Enemy(DEFAULT_ENEMY.positionX, firstRowEnemy);
+        allEnemies.push(enemy);
+    }
+
+    if (valueY === 2){
+        enemy = new Enemy(DEFAULT_ENEMY.positionX, secondRowEnemy);
+        allEnemies.push(enemy);
+    }
+
+    if (valueY === 3){
+        enemy = new Enemy(DEFAULT_ENEMY.positionX, thirdRowEnemy);
+        allEnemies.push(enemy);
+    }
```

Comment:
As a peer developer I am tasked to add two more rows to the play field.
Can we map `valueY` into enemy Y position?
Also is `valueY` is a pixel or field row?
Let the code say that.

Let's also `Array#map` `[1,2,3]` directly into `allEnemies`.

Reply:
> As a peer developer I am tasked to add two more rows to the play field. Can we map `valueY` into enemy Y position? Also is `valueY` is a pixel or field row? Let the code say that.
> 
> Let's also `Array#map` `[1,2,3]` directly into `allEnemies`.

done